gpu/gosl/examples/basic/main.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2022, Cogent Core. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This example just does some basic calculations on data structures and
+// reports the time difference between the CPU and GPU.
+package main
+
+import (
+	"embed"
+	"fmt"
+	"math/rand"
+	"runtime"
+	"unsafe"
+
+	"cogentcore.org/core/base/timer"
+	"cogentcore.org/core/gpu"
+)
+
+//go:generate ../../gosl cogentcore.org/core/math32/fastexp.go compute.go
```

Comment:
This (and other instances of the same pattern) results in failures for people who don't have `gosl` already built; how do we feel about that?